redwood-protocol-guest/src/commonMain/kotlin/app/cash/redwood/protocol/guest/GuestProtocolAdapter.kt

```diff
@@ -127,14 +127,44 @@ public interface GuestProtocolAdapter : EventSink {
   )
 
   @RedwoodCodegenApi
-  public fun appendRemove(
+  public abstract fun appendRemove(
     id: Id,
     tag: ChildrenTag,
     index: Int,
     count: Int,
-    removedIds: List<Id> = listOf(),
+    removedIds: List<Id>,
```

Comment:
This caused boxing in JS if the caller omitted this argument and relied on the default.

Reply:
Is there a way we can validate boxing doesn't occur in a test case? Or is it only possible when inspecting the generated code?

Reply:
I'll think about it. You definitely can see the boxing in a full compiler plugin, and maybe in FIR only. Probably also as a lint check.

Filed #2336